Random Users/Networking/UserClient.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  UserClient.swift
+//  Random Users
+//
+//  Created by ronald huston jr on 9/12/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import Foundation
+
+class UserClient {
+    
+    let baseURL = URL(string: "https://randomuser.me/api/?format=json&inc=name,email,phone,picture&results=1000")!
+    
+    func fetchUsers(completion: @escaping ([User]?, Error?) -> Void) {
+        URLSession.shared.dataTask(with: baseURL) { (data, _, error) in
+            if let error = error {
+                NSLog("error getting users: \(error)")
+            }
+            
+            guard let data = data else {
+                NSLog("no data returned from fetch")
+                completion(nil, error)
+                return
+            }
```

Comment:
What I like to do is hit the endpoint in Insomnia and figure out the correct successful response code, usually a 200 for a get request or 201 for a post, but always check these to be certain. In this case you a successful response code is 200, so something like this would work...
```suggestion
            if let response = response as? HTTPURLResponse, response.statusCode != 200 {
                print("Bad response code fetching random users: \(response)")
                completion(nil, error)
                return
            }
            guard let data = data else {
                NSLog("no data returned from fetch")
                completion(nil, error)
                return
            }
```
Here we are checking if we got a response back, if so down casting it as an HTTPURLResponse, then checking if that responses status code is anything other than 200, if it is it will early return out of the function.